### ISSUE TITLE
silx view does not interpret nxdata written by pni-libraries

### ISSUE THREAD

**Issue by jkotan** [silx-kit/silx#1773]
Hi,

It looks like `silx view` is not able to interpret nxdata in a file written by [python-pni](https://pni-libraries.github.io/python-pni/h5cpp_files.html), it cannot read string attributes (which are not written by h5py) so nxdata is not displayed as a plot. 

An example of the file written by `pni` can be found [here](https://github.com/jkotan/file-examples/blob/master/pninexus/mynewtest_00101.nxs).

Bests,
Jan

**Comment by vasole** [comment]
Your attributes are not strings but arrays of strings. It is a very common mistake we have observed on files generated using C++. Please make sure you conform to the NeXus specifications.

![image](https://user-images.githubusercontent.com/5636251/38606276-dc34154c-3d75-11e8-865b-afcf8659178b.png)

**Comment by vasole** [comment]
No, they are not conforming to the standard. Your files have to be fixed.

**Comment by jkotan** [comment]
http://download.nexusformat.org/doc/html/datarules.html#nexus-data-types

"NeXus accepts both variable and fixed length strings, as well as arrays of strings. Software that reads NeXus data files should support all of these.

Some file writers write strings as a string array of rank 1 and length 1. Clients should be prepared to handle such strings."

**Comment by t20100** [comment]
It should be possible to modify the `silx.io.nxdata._utils.get_attr_as_unicode` function to accommodate for this with e.g., an extra `output_format` argument to allow converting a list of string of length 1 to  string (and raise an error when the length of the list is not 1.

Contribution welcome.

**Comment by vasole** [comment]
strings

    NX_CHAR: The preferred string representation is UTF-8. Both fixed-length strings and variable-length strings are valid. String arrays cannot be used where only a string is expected (title, start_time, end_time, NX_class attribute,…). Fields or attributes requiring the use of string arrays will be clearly marked as such (like the NXdata attribute auxiliary_signals).

**Comment by t20100** [comment]
Sounds better.
The question remains whether we want the reading to be permissive or not....
Provided that this issue remains open for a while, it does not seems critical.
